setup.py

```diff
@@ -0,0 +1,132 @@
+"""
+setuptools-based build of ROOT.
+
+This script uses setuptools API to steer a custom CMake build of ROOT. All build
+configuration options are specified in the class responsible for building. A
+custom extension module is injected in the setuptools setup to properly generate
+the wheel with CPython extension metadata. Note that ROOT is first installed via
+CMake into a temporary directory, then the ROOT installation artifacts are moved
+to the final Python environment installation path, which often starts at
+${ENV_PREFIX}/lib/pythonXX.YY/site-packages, before being packaged as a wheel.
+"""
+
+import os
+import pathlib
+import shlex
+import subprocess
+import tempfile
+
+from setuptools import Extension, find_packages, setup
+from setuptools.command.build import build as _build
+from setuptools.command.install import install as _install
+
+# Get the long description from the README file
+SOURCE_DIR = pathlib.Path(__file__).parent.resolve()
+LONG_DESCRIPTION = (SOURCE_DIR / "README.md").read_text(encoding="utf-8")
+
+BUILD_DIR = tempfile.mkdtemp()
+INSTALL_DIR = tempfile.mkdtemp()
+
+# Name given to an internal directory within the build directory
+# used to mimick the structure of the target installation directory
+# in the user Python environment, usually named "site-packages"
+ROOT_BUILD_INTERNAL_DIRNAME = "mock_site_packages"
+
+
+class ROOTBuild(_build):
+    def run(self):
+        _build.run(self)
+
+        # Configure ROOT build
+        configure_command = shlex.split(
+            "cmake -GNinja -Dccache=ON "
+            "-Dgminimal=ON -Dasimage=ON -Dopengl=OFF "  # Graphics
+            "-Druntime_cxxmodules=ON -Drpath=ON -Dfail-on-missing=ON "  # Generic build configuration
+            "-Dtmva-pymva=OFF -Dtpython=OFF "  # Turn off explicitly components that link against libPython
+            "-Dbuiltin_nlohmannjson=ON -Dbuiltin_tbb=ON -Dbuiltin_xrootd=ON "  # builtins
+            "-Dbuiltin_lz4=ON -Dbuiltin_lzma=ON -Dbuiltin_zstd=ON -Dbuiltin_xxhash=ON "  # builtins
+            "-Dpyroot=ON -Ddataframe=ON -Dxrootd=ON -Dssl=ON -Dimt=ON "
+            "-Droofit=ON "
```

Comment:
```suggestion
            "-Droofit=ON -Dthisroot_scripts=OFF "
```
Something I remembered now: the `thisroot_scripts` are broken if the PYTHONDIR is not the LIBDIR, and I don't think they are necessary for the wheels.